doc: ActionDispatch::ServerTiming & rails_on_rack guide update [ci-skip] (#44186)
rails/rails: guides/source/rails_on_rack.md

Much of Action Controller's functionality is implemented as Middlewares. The following list explains the purpose of each of them:

**`ActionDispatch::HostAuthorization`**

> * Guards from DNS rebinding attacks by explicitly permitting the hosts a request can be sent to, and is passed the options set in `config.host_authorization`.

Reviewer: I realize this is [exactly what the API docs say](https://github.com/rails/rails/blob/8b2511b8ff28bc991ec459d798b8296195b423a3/actionpack/lib/action_dispatch/middleware/host_authorization.rb#L4-L6), but I feel like we can improve it a little for this guide:

```suggestion
* Guards from DNS rebinding attacks by explicitly permitting the hosts a request can be sent to. See the [configuration guide](configuring.html#actiondispatch-hostauthorization) for configuration instructions.
```

Author: Done

* Used for thread safe code reloading during development.

**`ActionDispatch::ServerTiming`**

> * Sets a Server-Timing header, containing one or more metrics and descriptions for the given request-response cycle.

Reviewer: Again, I realize this is [exactly what MDN says](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing), but I feel like we can tailor it a bit for our case:

```suggestion
* Sets a [`Server-Timing`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing) header containing performance metrics for the request.
```

Author: Done
